main.cpp

```diff
@@ -100,13 +103,388 @@ inf
+{
+    return divide(*x.value);
+}
+
+FloatType& FloatType::divide(const DoubleType& x)
+{
+    return divide(*x.value);
+}
+
+FloatType& FloatType::divide(const IntType& x)
+{
+    return divide(*x.value);
+}
+
+DoubleType& DoubleType::add(double x)
+{
+    *value += x;
+    return *this;
+}
+
+DoubleType& DoubleType::add(const FloatType& x)
+{
+    return add(*x.value);
+}
+
+DoubleType& DoubleType::add(const DoubleType& x)
+{
+    return add(*x.value);
+}
+
+DoubleType& DoubleType::add(const IntType& x)
+{
+    return add(*x.value);
+}
+
+DoubleType& DoubleType::subtract(double x)
+{
+    *value -= x;
+    return *this;
+}
+
+DoubleType& DoubleType::subtract(const FloatType& x)
+{
+    return subtract(*x.value);
+}
+
+DoubleType& DoubleType::subtract(const DoubleType& x)
+{
+    return subtract(*x.value);
+}
+
+DoubleType& DoubleType::subtract(const IntType& x)
+{
+    return subtract(*x.value);
+}
+
+DoubleType& DoubleType::multiply(double x)
+{
+    *value *= x;
+    return *this;
+}
+
+DoubleType& DoubleType::multiply(const FloatType& x)
+{
+    return multiply(*x.value);
+}
+
+DoubleType& DoubleType::multiply(const DoubleType& x)
+{
+    return multiply(*x.value);
+}
+
+DoubleType& DoubleType::multiply(const IntType& x)
+{
+    return multiply(*x.value);
+}
+
+DoubleType& DoubleType::divide(double x)
+{
+    if (x == 0.0)
+    {
+        std::cout << "warning: floating point division by zero!\n";
+    } 
+    *value /= x;
+    return *this;
+}
+
+DoubleType& DoubleType::divide(const FloatType& x)
+{
+    return divide(*x.value);
+}
+
+DoubleType& DoubleType::divide(const DoubleType& x)
+{
+    return divide(*x.value);
+}
+
+DoubleType& DoubleType::divide(const IntType& x)
+{
+    return divide(*x.value);
+}
+
+IntType& IntType::add(int x)
+{
+    *value += x;
+    return *this;
+}
+
+IntType& IntType::add(const FloatType& x)
+{
+    return add(*x.value);
+}
+
+IntType& IntType::add(const DoubleType& x)
+{
+    return add(*x.value);
+}
+
+IntType& IntType::add(const IntType& x)
+{
+    return add(*x.value);
+}
+
+IntType& IntType::subtract(int x)
+{
+    *value -= x;
+    return *this;
+}
+
+IntType& IntType::subtract(const FloatType& x)
+{
+    return subtract(*x.value);
+}
+
+IntType& IntType::subtract(const DoubleType& x)
+{
+    return subtract(*x.value);
+}
+
+IntType& IntType::subtract(const IntType& x)
+{
+    return subtract(*x.value);
+}
+
+IntType& IntType::multiply(int x)
+{
+    *value *= x;
+    return *this;
+}
+
+IntType& IntType::multiply(const FloatType& x)
+{
+    return multiply(*x.value);
+}
+
+IntType& IntType::multiply(const DoubleType& x)
+{
+    return multiply(*x.value);
+}
+
+IntType& IntType::multiply(const IntType& x)
+{
+    return multiply(*x.value);
+}
+
+IntType& IntType::divide(int x)
+{
+    if (x == 0)
+    {
+        std::cout << "error: integer division by zero is an error and will crash the program!\n";
+        return *this;
+    }
+    *value /= x;
+    return *this;
+}
```

Comment:
`return *this` is written twice. 
D.R.Y.
revise the logic so you only write it once.
- [ ] revise